onnx/defs/logical/defs.cc

```diff
@@ -121,6 +121,48 @@ ONNX_OPERATOR_SET_SCHEMA(
             {"tensor(bool)"},
             "Constrains output to boolean tensor."));
 
+ONNX_OPERATOR_SET_SCHEMA(
+    NE,
+    8,
+    OpSchema()
+        .FillUsing(BinaryLogicDocGenerator("not equal"))
+        .TypeConstraint(
+            "T",
+            {"tensor(bool)", "tensor(int32)", "tensor(int64)"},
+            "Constrains input to integral tensors.")
+        .TypeConstraint(
+            "T1",
+            {"tensor(bool)"},
+            "Constrains output to boolean tensor."));
+
+ONNX_OPERATOR_SET_SCHEMA(
+    LE,
+    8,
+    OpSchema()
+        .FillUsing(BinaryLogicDocGenerator("less than or equal"))
+        .TypeConstraint(
+            "T",
+            {"tensor(bool)", "tensor(int32)", "tensor(int64)"},
```

Comment:
What's the semantic of ‘<=' on bool?

Reply:
This is a good point. I wanted to stay consistent with numpy.less_equal functionality which would return a tensor filled with True. But I see the argument as to not supporting boolean tensors for this op.

Reply:
What we could do is to maintain support <= on boolean tensors (for consistency reasons), but write an optimization pass that would replace this comparison with a Boolean Tensor of True values.